From 9d394aaec3862766b45a78b55f552ddf854f2f18 Mon Sep 17 00:00:00 2001
From: Konrad Weihmann <kweihmann@outlook.com>
Date: Thu, 2 Jun 2022 10:25:54 +0200
Subject: [PATCH] setup.cfg: don't use ' in project URLs

Upstream-Status: Pending
Signed-off-by: Konrad Weihmann <kweihmann@outlook.com>
---
 setup.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.cfg b/setup.cfg
index 080d696..622daf8 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -32,7 +32,7 @@ keywords = static code analysis linter python lint
 project_urls = 
 	Homepage = https://www.pylint.org/
 	Source Code = https://github.com/PyCQA/pylint
-	What's New = https://pylint.pycqa.org/en/latest/whatsnew/
+	What is New = https://pylint.pycqa.org/en/latest/whatsnew/
 	Bug Tracker = https://github.com/PyCQA/pylint/issues
 	Discord Server = https://discord.com/invite/Egy6P8AMB5
 	Docs: User Guide = https://pylint.pycqa.org/en/latest/
-- 
2.25.1

