From a3f4a437133a8d973a962e94dd25626cf3ca5cea Mon Sep 17 00:00:00 2001
From: Konrad Weihmann <kweihmann@outlook.com>
Date: Sun, 10 May 2020 20:00:35 +0200
Subject: [PATCH] Makefile fixes

Signed-off-by: Konrad Weihmann <kweihmann@outlook.com>
Upstream-Status: Inappropriate [configuration]
---
 Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/Makefile b/Makefile
index 94e6df241..30fe68f6a 100644
--- a/Makefile
+++ b/Makefile
@@ -122,11 +122,11 @@ else ifeq ($(CXX), c++)
 endif
 
 ifeq ($(HAVE_RULES),yes)
-    override CXXFLAGS += -DHAVE_RULES -DTIXML_USE_STL $(shell pcre-config --cflags)
+    override CXXFLAGS += -DHAVE_RULES -DTIXML_USE_STL $(shell pkg-config libpcre --cflags)
     ifdef LIBS
-        LIBS += $(shell pcre-config --libs)
+        LIBS += $(shell pkg-config libpcre --libs)
     else
-        LIBS=$(shell pcre-config --libs)
+        LIBS=$(shell pkg-config libpcre --libs)
     endif
 endif
 
-- 
2.20.1

