From 7e8e50f734aa6bf67b1c242b16fdb6e6b18b221c Mon Sep 17 00:00:00 2001
From: Konrad Weihmann <kweihmann@outlook.com>
Date: Sun, 1 Sep 2019 20:18:17 +0200
Subject: [PATCH] Makefile fixes

---
 Makefile | 24 +++++++++++++-----------
 1 file changed, 13 insertions(+), 11 deletions(-)

diff --git a/Makefile b/Makefile
index 5c9539ff2..17aad9d7b 100644
--- a/Makefile
+++ b/Makefile
@@ -106,17 +106,19 @@ else ifeq ($(CXX), c++)
 endif
 
 ifeq ($(HAVE_RULES),yes)
-    override CXXFLAGS += -DHAVE_RULES -DTIXML_USE_STL $(shell pcre-config --cflags)
+    override CXXFLAGS += -DHAVE_RULES -DTIXML_USE_STL $(shell pkg-config libpcre --cflags)
     ifdef LIBS
-        LIBS += $(shell pcre-config --libs)
+        LIBS += $(shell pkg-config libpcre --libs)
     else
-        LIBS=$(shell pcre-config --libs)
+        LIBS=$(shell pkg-config libpcre --libs)
     endif
 endif
 
-ifndef PREFIX
-    PREFIX=/usr
-endif
+DESTDIR?= ""
+prefix?= "/usr"
+bindir?= "$(prefix)/bin"
+BINDIR=$(bindir)
+PREFIX=$(prefix)
 
 ifndef INCLUDE_FOR_LIB
     INCLUDE_FOR_LIB=-Ilib -isystem externals -isystem externals/simplecpp -isystem externals/tinyxml
@@ -130,8 +132,6 @@ ifndef INCLUDE_FOR_TEST
     INCLUDE_FOR_TEST=-Ilib -Icli -isystem externals/simplecpp -isystem externals/tinyxml
 endif
 
-BIN=$(DESTDIR)$(PREFIX)/bin
-
 # For 'make man': sudo apt-get install xsltproc docbook-xsl docbook-xml on Linux
 DB2MAN?=/usr/share/sgml/docbook/stylesheet/xsl/nwalsh/manpages/docbook.xsl
 XP=xsltproc -''-nonet -''-param man.charmap.use.subset "0"
@@ -301,9 +301,11 @@ tags:
 	ctags -R --exclude=doxyoutput --exclude=test/cfg --exclude=test/synthetic cli externals gui lib test
 
 install: cppcheck
-	install -d ${BIN}
-	install cppcheck ${BIN}
-	install htmlreport/cppcheck-htmlreport ${BIN}
+	install -d ${DESTDIR}${BINDIR}
+	install cppcheck ${DESTDIR}${BINDIR}
+	install addons/*.py ${DESTDIR}${BINDIR}
+	install addons/*/*.py ${DESTDIR}${BINDIR}
+	install htmlreport/cppcheck-htmlreport ${DESTDIR}${BINDIR}
 ifdef FILESDIR
 	install -d ${DESTDIR}${FILESDIR}
 	install -d ${DESTDIR}${FILESDIR}/addons
-- 
2.17.1

