From a72112621619613c0449fea352732aa2955da5d4 Mon Sep 17 00:00:00 2001
From: Konrad Weihmann <kweihmann@outlook.com>
Date: Sat, 9 May 2020 21:31:40 +0200
Subject: [PATCH] remove protobuf requirement

Signed-off-by: Konrad Weihmann <kweihmann@outlook.com>
Upstream-Status: Inappropriate [disable-feature]
---
 configure.ac | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git a/configure.ac b/configure.ac
index 7a9bb6c2..f0be01c1 100644
--- a/configure.ac
+++ b/configure.ac
@@ -166,23 +166,6 @@ AC_ARG_ENABLE([debug-dex],
     CFLAGS="$CFLAGS -DDEBUG_DEX_MODULE"
   fi])
 
-AC_ARG_ENABLE([pb-tests],
-  [AS_HELP_STRING([--enable-pb-tests], [enable protobuf test module])],
-  [if test x$enableval = xyes; then
-    build_pb_tests_module=true
-    AC_CHECK_PROG(PROTOC, protoc, protoc)
-    AS_IF([test "x${PROTOC}" == "x"],
-        [AC_MSG_ERROR([protobuf compiler "protoc" not found])])
-    # AC_CHECK_PROG(PROTOC_GEN_YARA, protoc-gen-yara, protoc-gen-yara)
-    # AS_IF([test "x${PROTOC_GEN_YARA}" == "x"],
-    #    [AC_MSG_ERROR([please install https://github.com/VirusTotal/protoc-gen-yara])])
-    PKG_CHECK_MODULES(PROTOBUF_C, libprotobuf-c >= 1.0.0)
-    AC_CHECK_LIB(protobuf-c, protobuf_c_message_unpack,,
-      AC_MSG_ERROR([please install libprotobuf-c library]))
-    CFLAGS="$CFLAGS -DPB_TESTS_MODULE"
-    PC_REQUIRES_PRIVATE="$PC_REQUIRES_PRIVATE libprotobuf-c"
-  fi])
-
 AC_ARG_WITH([jemalloc],
   [AS_HELP_STRING([--with-jemalloc],
     [use jemalloc to debug heap-related issues])],
-- 
2.20.1

