From 9d74d2de6a23849536f858f69c1c1b933123bbc2 Mon Sep 17 00:00:00 2001
From: Konrad Weihmann <kweihmann@outlook.com>
Date: Wed, 10 Feb 2021 20:37:20 +0100
Subject: [PATCH] remove version binding for typing

Upstream-Status: Inappropriate [oe-specific]
Signed-off-by: Konrad Weihmann <kweihmann@outlook.com>
---
 setup.cfg | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.cfg b/setup.cfg
index 645cdbb..e5ac6c0 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -39,7 +39,7 @@ install_requires =
     requests >= 2.7.0, < 3.0.0
     intbitset >= 2.3.0, < 3.0
     saneyaml
-    typing >=3.6, < 3.7; python_version < "3.7"
+    typing >=3.6
 
 setup_requires = setuptools_scm[toml] >= 4
 
-- 
2.25.1

