From ac882a7ab4988a4cd80e81863c11cdad20374a91 Mon Sep 17 00:00:00 2001
From: Konrad Weihmann <kweihmann@outlook.com>
Date: Sun, 15 Nov 2020 13:35:39 +0100
Subject: [PATCH] Remove console entry script

Signed-off-by: Konrad Weihmann <kweihmann@outlook.com>
Upstream-Status: Inappropriate [disable-feature]
---
 setup.py | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/setup.py b/setup.py
index 617f7bb..fa2064f 100755
--- a/setup.py
+++ b/setup.py
@@ -41,12 +41,6 @@ setup(
     package_dir={'': 'src'},
     packages=['ninja'],
 
-    entry_points={
-        'console_scripts': [
-            'ninja=ninja:ninja'
-        ]
-    },
-
     url=r'http://ninja-build.org/',
     download_url=r'https://github.com/ninja-build/ninja/releases',
 
